File: src/main/java/com/clinica/atendimento/model/enums/Sexo.java
package com.clinica.atendimento.model.enums;

import lombok.Getter;

@Getter
public enum Sexo {
    FEMININO("F"),
    MASCULINO("M");

    private final String codigo;

    Sexo(String codigo) {
        this.codigo = codigo;
    }

    public static Sexo fromCodigo(String codigo) {
        if (codigo == null) {
            return null;
        }

        for (Sexo sexo : Sexo.values()) {
            if (sexo.getCodigo().equals(codigo)) {
                return sexo;
            }
        }
        throw new IllegalArgumentException("Código de sexo inválido: " + codigo);
    }
}

File: src/main/kotlin/com/clinica/atendimento/model/enums/Sexo.kt
package com.clinica.atendimento.model.enums

enum class Sexo(val codigo: String) {
    FEMININO("F"),
    MASCULINO("M");

    companion object {
        @JvmStatic
        fun fromCodigo(codigo: String?): Sexo? {
            if (codigo == null) {
                return null
            }

            for (sexo in values()) {
                if (sexo.codigo == codigo) {
                    return sexo
                }
            }
            throw IllegalArgumentException("Código de sexo inválido: $codigo")
        }
    }
}